docs/TpchBenchmark.md

```diff
@@ -0,0 +1,453 @@
+
+## Summary
+If a use of Velox matches the use case of the TcphBenchmark then it is a good tool to test, I/O and driver performance for specific TCP-H queries. This would benefit execution of specific production workloads that are like the chosen queries. If in multi-process use case, like Spark/Gluten/Velox configuration, the recommendation is to oversubscribe I/O threads between 2X and 3X vCPUs and tune the 2 coalesce configurations exposed.
+
+---
+## Appendix A: Patch to Build TpchBenchmark Executable
+```diff
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index f43b8618..a777bbb8 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -40,7 +40,7 @@ message(STATUS "Build type: ${CMAKE_BUILD_TYPE}")
+ option(
+   VELOX_BUILD_TESTING
+   "Enable Velox tests. This will enable all other build options automatically."
+-  ON)
++  OFF)
+ option(
+   VELOX_BUILD_MINIMAL
+   "Build a minimal set of components only. This will override other build options."
+@@ -70,13 +70,13 @@ option(VELOX_ENABLE_EXAMPLES
+        "Build examples. This will enable VELOX_ENABLE_EXPRESSION automatically."
+        OFF)
+ option(VELOX_ENABLE_SUBSTRAIT "Build Substrait-to-Velox converter." OFF)
+-option(VELOX_ENABLE_BENCHMARKS "Enable Velox top level benchmarks." OFF)
++option(VELOX_ENABLE_BENCHMARKS "Enable Velox top level benchmarks." ON)
+ option(VELOX_ENABLE_BENCHMARKS_BASIC "Enable Velox basic benchmarks." OFF)
+-option(VELOX_ENABLE_S3 "Build S3 Connector" OFF)
++option(VELOX_ENABLE_S3 "Build S3 Connector" ON)
+ option(VELOX_ENABLE_HDFS "Build Hdfs Connector" OFF)
+-option(VELOX_ENABLE_PARQUET "Enable Parquet support" OFF)
++option(VELOX_ENABLE_PARQUET "Enable Parquet support" ON)
+ option(VELOX_ENABLE_ARROW "Enable Arrow support" OFF)
+-option(VELOX_ENABLE_CCACHE "Use ccache if installed." ON)
++option(VELOX_ENABLE_CCACHE "Use ccache if installed." OFF)
+ 
+ option(VELOX_BUILD_TEST_UTILS "Builds Velox test utilities" OFF)
+ option(VELOX_BUILD_PYTHON_PACKAGE "Builds Velox Python bindings" OFF)
+diff --git a/Makefile b/Makefile
+index e21f6b33..9578c377 100644
+--- a/Makefile
++++ b/Makefile
+@@ -80,7 +80,7 @@ cmake:					#: Use CMake to create a Makefile build system
+ 		${EXTRA_CMAKE_FLAGS}
+ 
+ build:					#: Build the software based in BUILD_DIR and BUILD_TYPE variables
+-	cmake --build $(BUILD_BASE_DIR)/$(BUILD_DIR) -j $(NUM_THREADS)
++	cmake --build $(BUILD_BASE_DIR)/$(BUILD_DIR) -j $(NUM_THREADS) --target velox_tpch_benchmark
+ 
+ debug:					#: Build with debugging symbols
+ 	$(MAKE) cmake BUILD_DIR=debug BUILD_TYPE=Debug
+diff --git a/velox/benchmarks/tpch/CMakeLists.txt b/velox/benchmarks/tpch/CMakeLists.txt
+index fc865154..a8a7e573 100644
+--- a/velox/benchmarks/tpch/CMakeLists.txt
++++ b/velox/benchmarks/tpch/CMakeLists.txt
+@@ -27,6 +27,7 @@ target_link_libraries(
+   velox_hive_connector
+   velox_exception
+   velox_memory
++  velox_s3fs
+   velox_process
+   velox_serialization
+   velox_encode
+```
+
+---
+## Appendix B: TpchBenchmark Tool Help Output
```

Comment:
Would be also nice if we have an appendix with our current results and the test setup.